WordPress/Classes/Networking/PlansRemote.swift

```diff
@@ -54,9 +56,173 @@ private func mapPlansResponse(response: AnyObject) throws -> (activePlan: Plan,
             return (result.0, plans)
         }
     })
+    
     guard let activePlan = parsedResponse.0 else {
         throw PlansRemote.Error.NoActivePlan
     }
     let availablePlans = parsedResponse.1.sort()
     return (activePlan, availablePlans)
 }
+
+private func parseFeatureGroups(json: [[String: AnyObject]], forPlan plan: Plan) {
+    let groups: [PlanFeatureGroup] = json.flatMap { groupJson in
+        guard let slugs = groupJson["items"] as? [String] else { return nil }
+        return PlanFeatureGroup(title: groupJson["title"] as? String, slugs: slugs)
+    }
+    
+    PlanFeatureGroup.setGroups(groups, forPlan: plan)
+}
+
+class PlanFeaturesRemote: ServiceRemoteREST {
+    
+    enum Error: ErrorType {
+        case DecodeError
+    }
+    
+    private var cacheDate: NSDate?
+    private let cacheFilename = "plan-features.json"
+    
+    func getPlanFeatures(success: PlanFeatures -> Void, failure: ErrorType -> Void) {
+        // First we'll try and return plan features from memory, then check our disk cache,
+        // and finally hit the network if we don't have anything recent enough
+        if let planFeatures = inMemoryPlanFeatures {
```

Comment:
Just leaving a note that I think the cache system should be extracted from here and made generic, although not on this PR. I started working on a similar [cache implementation](https://gist.github.com/koke/a57b48a032e26fca6ac0) recently, although it doesn't handle expiration yet.

Also, ideally we'd use NSURL caching but that will require some API changes.